src/main/java/pt/tqs/g205/services/CategoriaPratoService.java

```diff
@@ -0,0 +1,46 @@
+package pt.tqs.g205.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import pt.tqs.g205.domain.CategoriaPrato;
+import pt.tqs.g205.repositories.CategoriaPratoRepository;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/**
+ * Servicos para a classe CategoriaPrato.
+ */
+@Service
+public class CategoriaPratoService {
+
+  @Autowired
+  private CategoriaPratoRepository categoriaRepo;
+  
+  /**
+   * Obter todas as categorias de pratos.
+   * @return lista de categorias.
+   */
+  public List<CategoriaPrato> getAll() {
+    List<CategoriaPrato> categorias = categoriaRepo.findAll();
+    categorias.forEach(e -> e.setPratos(null));
+    return categorias;
+  }
+  
+  /**
+   * Obter uma categoria pelo seu id.
+   * @param id id da categoria.
+   * @return categoria.
+   */
+  public CategoriaPrato getById(Integer id) throws NoSuchElementException {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the declaration of thrown exception 'java.util.NoSuchElementException' which is a runtime exception. ([squid:RedundantThrowsDeclarationCheck](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3ARedundantThrowsDeclarationCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=5&issues=AWNDDljGXOBV_Yrh-1IU&open=AWNDDljGXOBV_Yrh-1IU)

[ISSUE_KEY]: AWNDDljGXOBV_Yrh-1IU